### PR TITLE
Added support for metronome dashboard

### PR DESCRIPTION
**Pull request by rokso**
- Added index for `ConvertMetToEth`, `ConvertEthToMet` and `LogAuctionFundsIn` event.
- Added new routers to support dashboard APIs
- Dashboard APIs will serve market stats, Auction stats and Autonomous Converter stats

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -6,44 +6,30 @@ const Server = require('./server')
...
-      db: this.database,
-      ethApi: this.ethApi,
-      socket: this.socket
-    })
-
-    this.statusCollector = StatusCollector.start()
+    this.tokenExporter = new TokenExporter(this.database, this.socket)
+    this.accExporter = new ACCExporter(this.database, this.socket)
+    // this.status = new Status(this.database, this.socket)
+    // this.statusCollector = StatusCollector.start()
```
Comment (patidarmanoj10) [inline lib/index.js]: @rokso  we will need to uncomment and work as is.

Comment (rokso) [reply lib/index.js]: There are 2 commits can you more sure, you are checking latest changes.

**Comment by patidarmanoj10** [comment]
@rokso   There are two commits and two authors.  if you squash will show my commit only.  is there any way we squash these two and show your commit? I guess , its fine if you fork again and send pr with your commit.

**Comment by rokso** [comment]
> @rokso There are two commits and two authors. if you squash will show my commit only. is there any way we squash these two and show your commit? I guess , its fine if you fork again and send pr with your commit.

Fixed it :)

File: lib/index.js (hunk 1 above)
Comment (rokso) [reply lib/index.js]: Resolved, as I end up squashing multi author commits.

File: lib/market-data.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                return {
+                    symbol: data.symbol,
+                    value: parseFloat(data.priceUsd, 10),
+                    supply: data.supply ? Math.round(data.supply) : null,
+                    marketCapUsd: data.marketCapUsd
+                }
+            }
+        })
+        .catch(err => {
+            // eslint-disable-next-line no-console
```
Comment (gabmontes) [inline lib/market-data.js]: This ESLint comment is no longer needed.

File: route/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+const db = require('../lib/stats-db')
+const converterStats = require('./stats/converter-stats')
+const auctionsStats = require('./stats/auction-stats')
+const moment = require('moment-timezone')
+const router = new Router()
+
+function parseFromTo(req, res, next) {
+    moment.tz.setDefault('UTC')
+    const { query } = req
+    const birthTime = 1529280000 // in seconds
```
Comment (gabmontes) [inline route/dashboard.js]: Perhaps is better to have this in the config.

File: route/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        const stats = {}
+        stats.price = price
+        stats.availability = data.minting
+        stats.totalAuctions = data.currAuction
+
+        const nextMidnight = moment().startOf('day').add(1, 'day')
+        const remainingSeconds = nextMidnight.diff(moment(), 'seconds')
+        const duration = moment.duration(remainingSeconds, 'seconds')
+
+        stats.timeRemaining = `${duration.hours()} hour(s) ${duration.minutes()} minute(s) ${duration.seconds()} second(s)`
```
Comment (gabmontes) [inline route/dashboard.js]: `timeRemaining` should be provided in raw seconds/ms too.

File: route/stats/auction-stats.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+ * Get function to get unique ISA (Initial Supply Auction) and daily buyers.
+ * 
+ * @param {object} model Mongo model to run query over database.
+ * @param {object} params Contains 'from' and 'to' for query.
+ * @returns {BuyerCount} Unique buyer count
+ */
+function getBuyerCount(model, params) {
+    const { from, to } = params
+
+    const isaEndTime = 1529884800
```
Comment (gabmontes) [inline route/stats/auction-stats.js]: Move to config?

File: lib/market-data.js (hunk 2 above)
Comment (rokso) [reply lib/market-data.js]: Bummer, good catch.

File: route/dashboard.js (hunk 3 above)
Comment (rokso) [reply route/dashboard.js]: Good idea.

File: route/dashboard.js (hunk 4 above)
Comment (rokso) [reply route/dashboard.js]: Yes, I will change it.

**Pull request merged**